From 2f483d1dc3d98ec082f597e176706ed40bf8421d Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Thu, 28 Mar 2024 12:27:51 +0100
Subject: [PATCH 1006/1141] pinctrl: stm32: Manage irq affinity settings

Trying to set the affinity of the interrupt associated to stm32
pinctrl results in a write error because stmf32 pinctrl was not
managing the set of the irq smp affinity. It is resolved by
adding the .irq_set_affinity to the stm32_gpio_irq_chip struct and
fill it with a function to manage affinity.

Change-Id: I32cb2ca38936f114874abc404cab76bc64f93c14
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368513
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/pinctrl/stm32/pinctrl-stm32.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/pinctrl/stm32/pinctrl-stm32.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32.c
@@ -602,6 +602,8 @@ static struct irq_chip stm32_gpio_irq_ch
 	.irq_set_wake	= irq_chip_set_wake_parent,
 	.irq_request_resources = stm32_gpio_irq_request_resources,
 	.irq_release_resources = stm32_gpio_irq_release_resources,
+	.irq_set_affinity = IS_ENABLED(CONFIG_SMP) ? irq_chip_set_affinity_parent : NULL,
+
 };
 
 static int stm32_gpio_domain_select(struct irq_domain *dm,
